gettsim/synthetic.py

```diff
@@ -0,0 +1,303 @@
+import datetime
+import itertools
+
+import numpy as np
+import pandas as pd
+
+from gettsim.policy_environment import _load_parameter_group_from_yaml
+
+
+def append_other_hh_members(
+    df, hh_typ, n_children, age_adults, age_children, double_earner
+):
+    """
+    duplicates information from the one person already created
+    as often as needed and adjusts columns that differ
+    """
+    # create empty dataframe with correct columns and datatypes
+    new_df = df.iloc[0:0].copy()
+    if hh_typ == "single" and n_children == 0:
+        return None
+
+    # If couple, create an additional adult
+    if hh_typ == "couple":
+        adult = df.copy()
+        adult["alter"] = age_adults[1]
+        if not double_earner:
+            adult["bruttolohn_m"] = 0
+
+        new_df = new_df.append(adult, ignore_index=True)
+
+    child = df.copy()
+    child["kind"] = True
+    child["in_ausbildung"] = True
+    child["bruttolohn_m"] = 0
+
+    if n_children == 1:
+        children = child.copy()
+        children["alter"] = age_children[0]
+    elif n_children == 2:
+        children = child.append(child, ignore_index=True)
+        children["alter"] = age_children
+    else:
+        children = None
+
+    # append children
+    new_df = new_df.append(children)
+    new_df["tu_vorstand"] = False
+
+    return new_df
+
+
+def create_synthetic_data(
+    hh_typen=None,
+    n_children=None,
+    age_adults=None,
+    age_children=None,
+    baujahr=1980,
+    double_earner=False,
+    policy_year=datetime.datetime.now().year,
+    heterogeneous_vars=(),
+    **kwargs,
+):
+    """
+    Creates a dataset with hypothetical household types,
+    which can be used as input for gettsim
+
+    hh_typen (list of str):
+        Allowed Household Types: 'single', 'couple'
+
+    n_children (list of int):
+        number of children
+
+    age_adults (list of int):
+        Assumed age of adult(s)
+
+    age_children (list of int):
+        Assumed age of children (first and second child, respectively)
+
+    baujahr (int):
+        Construction year of building
+
+    double_earner (bool):
+        whether or not both adults should be assigned the same value for 'bruttolohn_m'
+
+    heterogenous_vars (dict):
+        if specified, contains the variable name as key and a list of values
+
+    policy_year (int):
+        the year from which the reference data on housing are drawn.
+
+    kwargs:
+
+    bruttolohn_m, kapital_eink_m, eink_selbst_m, vermögen_hh (int):
+        values for income and wealth, respectively.
+        only valid if heterogenous_vars is empty
+    """
+    # Set Defaults
+    if hh_typen is None:
+        hh_typen = ["single", "couple"]
+    if n_children is None:
+        n_children = [0, 1, 2]
+    if age_adults is None:
+        age_adults = [35, 35]
+    if age_children is None:
+        age_children = [3, 8]
+    # Check inputs
+    for t in hh_typen:
```

Comment:
Please also check if `hh_typen` and `n_children` is entered as list/collection and give a meaningful error message if not. Especially for `n_children`, it's not necessarily intuitive that a single number doesn't work. Maybe in this case, we should convert the single int to a one element list.